02-Calculator/Calculator.js

```diff
@@ -3,26 +3,130 @@ import {MyMath} from "../01-MyMath/MyMath.js";
 export class Calculator {
 
     constructor(numpad, outputCalculation, outputSolution) {
+        this.numpad = numpad;
+        this.outputCalculation = outputCalculation;
+        this.outputSolution = outputSolution;
+
+        this.mymath = new MyMath(0);
+        this.last_operation = (value => this.mymath.add(value));
+
+        this.setupNumPad();
     }
 
     setupNumPad() { 
+        //generiere Ziffernblock von 1 bis 9
+        for(let row = 2; row + 1; row--)
+        {
+            let new_div = document.createElement("div");
+            new_div.classList.add("flex-4");
+            for(let i = 1; i <= 3; i++)
+            {
+                let new_button = document.createElement("button");
+                let button_num = row*3 + i;
+                new_button.innerText = "" + button_num;
+                new_div.appendChild(new_button);
+            }
+            this.numpad.appendChild(new_div);
+        }
+        
```

Comment:
'flex-4' macht deine doppelte for schleife redundant, da flex-4 dafür sorgt dass sich elemente sowieso in 4er Reihen anordnen.
Warum 4er reihen? Hier ist eigtl geplant dass du die Operatoren in der gleichen Schleife hinzufügst.
In der Art: ["x<sup>y</sup>","x!","DEL","AC",7,8,9,"/",4,5,6,"*",1,2,3,"-",0,".","+","="].forEach()
Dann sparst du dir die ganze row logik und muss nicht für operatoren und zahlen zweimal den gleichen code haben um sie zu generieren.
Der Rest deines Codes wär damit auch kompatibel.